lib/repl.js

```diff
@@ -689,8 +690,71 @@ function intFilter(item) {
   return /^[A-Za-z_$]/.test(item);
 }
 
+const defaultProperties = [
+  [ Array, Object.getOwnPropertyNames([]).filter(intFilter) ],
+  [ Buffer, Object.getOwnPropertyNames(Buffer.alloc(0)).filter(intFilter) ],
+
+  [ Uint8Array,
+    Object.getOwnPropertyNames(new Uint8Array()).filter(intFilter) ],
+  [ Uint16Array,
+    Object.getOwnPropertyNames(new Uint16Array()).filter(intFilter) ],
+  [ Uint32Array,
+    Object.getOwnPropertyNames(new Uint32Array()).filter(intFilter) ],
+  [
+    Uint8ClampedArray,
+    Object.getOwnPropertyNames(new Uint8ClampedArray()).filter(intFilter) ],
+  [ Int8Array,
+    Object.getOwnPropertyNames(new Int8Array()).filter(intFilter) ],
+  [ Int16Array,
+    Object.getOwnPropertyNames(new Int16Array()).filter(intFilter) ],
+  [ Int32Array,
+    Object.getOwnPropertyNames(new Int32Array()).filter(intFilter) ],
+  [ Float32Array,
+    Object.getOwnPropertyNames(new Float32Array()).filter(intFilter) ],
+  [ Float64Array,
+    Object.getOwnPropertyNames(new Float64Array()).filter(intFilter) ]
+];
+const ARRAY_LENGTH_THRESHOLD = 1e6;
+
+function mayBeLargeObject(obj) {
+  // `Buffer.prototype` passes the `Buffer.isBuffer` and
+  // `instanceof Uint8Array`.
+  //
+  // Refs: https://github.com/nodejs/node/pull/11961
```

Comment:
This will still throw an error on e.g. `Object.create(Uint8Array.prototype)`.

Instead of all the checks below, it could be implemented simply and reliably as:

```js
const { isTypedArray } = process.binding('util');

function maybeLargeObject(obj) {
  return Array.isArray(obj) || isTypedArray(obj);
}
```

(Buffers pass the `isTypedArray` test.)